iothub/device/src/FileUpload/FileUploadCompletionNotification.cs

```diff
@@ -10,10 +10,17 @@ namespace Microsoft.Azure.Devices.Client.Transport
     /// </summary>
     public class FileUploadCompletionNotification
     {
+        /// <summary>
+        /// Initialize an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        public FileUploadCompletionNotification()
```

Comment:
```suggestion
        protected internal FileUploadCompletionNotification()
```

Reply:
If it is for unit testing only, make it protected internal so it can be called by our tests, or by end-users who need to mock it.